src/lib/consent-components/consent.module.ts

```diff
@@ -19,21 +23,39 @@ import { ThemeDirective } from './theme/theme.directive';
     ConsentModalComponent,
     ConsentModalNavigationComponent,
     MarkdownToHtmlComponent,
-    ThemeDirective
+    ThemeDirective,
   ],
   imports: [
     CommonModule,
     NgxSmartModalModule.forRoot(),
     UiSwitchModule,
-    MarkdownModule.forRoot()
+    MarkdownModule.forRoot(),
   ],
   exports: [
     ConsentButtonComponent,
     ConsentSwitchComponent,
     ConsentModalComponent,
     ConsentModalNavigationComponent,
     MarkdownToHtmlComponent,
-    ThemeDirective
-  ]
+    ThemeDirective,
+  ],
 })
-export class ConsentModule { }
+export class ConsentModule {
+  public static forRoot(config: ConsentModuleConfig): ModuleWithProviders {
+    return {
+      ngModule: ConsentModule,
+      providers: [
+        MarkdownService,
+        {
+          provide: ConsentModuleConfigService,
+          useValue: config,
+        },
+        SmartlookService,
+        {
+          provide: ConsentModuleConfigService,
```

Comment:
Why do you provide ConsentModuleConfigService twice?

Reply:
It's a token for injecting config files to other services

Reply:
```
{
          provide: ConsentModuleConfigService,
          useValue: config,
 }
...
{
          provide: ConsentModuleConfigService,
          useValue: config,
 },
```
You specify twice that angular should use config for ConsentModuleConfigService. Once would be enough.

Reply:
ConsentModuleConfigService is a injection token, not real servise.
export const ConsentModuleConfigService = new InjectionToken<ConsentModuleConfig>("ConsentModuleConfig");
It providesthe same config from forRoot() to MarkdownService and SmartlookService
https://angular.io/guide/dependency-injection-providers#non-class-dependencies

Reply:
I'll change the name of token because it's confusing